.github/workflows/pr-validation.yml

```diff
@@ -0,0 +1,157 @@
+          # key: ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-${{ needs.get-variables.outputs.commitsha }}
+          # restore-keys: |
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-${{ needs.get-variables.outputs.shortref }}-workspace-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-${{ hashFiles('**/yarn.lock') }}-branch-
+          #   ${{ runner.os }}-node-${{ matrix.node-version }}-yarn-
+          #   ${{ runner.os }}-
+
+      - name: Install project dependencies
+        run: yarn install --frozen-lockfile --non-interactive
+
+      # Is 'yarn pack --filename ./dist/ngxs-core.tgz' needed during premerge checks?
+      - name: Compile NGXS and create a pack
+        run: |
+          yarn build
+          yarn pack --filename ./dist/ngxs-core.tgz
+
+      - name: Lint - TSLint
+        run: yarn lint
+
+      - name: Lint - ESLint
+        run: yarn eslint
+
+      - name: Unit Tests
+        run: yarn test:ci
+
+      ## This step was commented in the CircleCI workflows config.
+      # - name: Integration Tests
+      #   run: yarn test:ci:integration
+
+      ## This step was commented in the CircleCI workflows config.
+      # - name: E2E Tests
+      #   run: yarn test:ci:e2e
+
+      ## This step was commented in the CircleCI workflows config.
+      # - name: Integration Tests - SSR
+      #   run: yarn test:ci:integration:ssr
+
+      - name: Integration Tests - ng7
+        run: yarn integration:ng7
+
+      - name: Integration Tests - ng8
+        run: yarn integration:ng8
+
+      - name: Integration Tests - ng9, Ivy off
+        run: yarn integration:ng9:ivy:off
+
+      - name: Integration Tests - ng9, Ivy
+        run: yarn integration:ng9:ivy
+
+      - name: Integration Tests - ng10, Ivy off
+        run: yarn integration:ng10:ivy:off
+
+      - name: Integration Tests - ng10, Ivy
+        run: yarn integration:ng10:ivy
+
+      - name: Integration Tests - ng11, Ivy off
+        run: yarn integration:ng11:ivy:off
+
+      - name: Integration Tests - ng11, Ivy
+        run: yarn integration:ng11:ivy
+
+      - name: Integration Tests - ng12, Ivy
+        run: yarn integration:ng12:ivy
+
+      - name: Integration Tests - ng13, Ivy
+        run: yarn integration:ng13:ivy
+
+      - name: Integration Tests - types
+        run: yarn test:types
+
+      - name: Check the size of NGXS bundle
+        run: yarn bundlesize
+
+      ## Actually, this is probably not needed during premerge checks.
+      ## It makes sense to store test results only after testing already code merged into the trunk.
+      ## It does not seem to make sense to upload test results after each push to the branch that has not need merged into the trunk, there will be a lot of useless artifacts on Code Climate.
```

Comment:
I think that there is value to push the coverage to code climate so that the code climate checks are run as part of the PR checks, and so that the code climate report can be inspected for the PR.

Reply:
uncommented this step